### PR TITLE
Adds property type to taxonomy popups where available

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

property-type-in-popup

I found myself wanting to know the type of properties while selecting them

## How did you test this code?

By running it locally and viewing property selection pop-ups with and without the collaboration-taxonomy flag

### CODE REVIEW

**Review by clarkus — CHANGES_REQUESTED** [review]
We're accounting for this in the taxonomy card updates, though I don't know if it has been shipped or not yet. I originally had the property type in the header adjacent to the property name, but I can see how that could interfere with showing names with higher character counts. 

Screen Shot 2022-02-14 at 9 58 51 AM

How about we put it adjacent to the `sent as` value without wrapping to a new line?
Screen Shot 2022-02-14 at 10 01 50 AM

File: frontend/src/lib/components/PropertiesTable.scss [hunk 1]
```diff
@@ -25,6 +25,15 @@
...
     }
+
+    /*
+        generates .item-popup .property-value-type
+        only needed until feature flag collaborations-taxonomy is removed
+    */
+    @at-root .item-popup & {
+        margin-left: -1px;
+        margin-top: 0.5em;
+    }
```
Comment (alexkim205) [inline frontend/src/lib/components/PropertiesTable.scss:36]: nit: I think it'd be ideal if we can avoid injecting styles into the root. Is there a way to make this selector more specific (either with a more specific selector or placing it in-line js style)

File: frontend/src/lib/components/DefinitionPopup/DefinitionPopup.tsx [hunk 2]
```diff
@@ -441,6 +453,7 @@ export const renderItemPopup = (
                 <DefinitionPopup.HorizontalLine />
                 {/* Things start to get different here */}
                 {renderRestOfDefinition(item, listGroupType, hasTaxonomyFeatures)}
+                <DefinitionPopup.Type item={item as PropertyDefinition} />
```
Comment (alexkim205) [inline frontend/src/lib/components/DefinitionPopup/DefinitionPopup.tsx:456]: Can we place this inside of the renderRestOfDefinition method? I think it'd be cleaner to render this only when item is a property definition.

File: frontend/src/lib/components/DefinitionPopup/DefinitionPopup.tsx [hunk 3]
```diff
@@ -228,9 +228,21 @@ function Card({ title, value }: { title: string; value: React.ReactNode }): JSX.
     )
 }
 
+function Type({ item }: { item: PropertyDefinition | null }): JSX.Element {
+    console.log({ item, pt: item?.property_type })
```
Comment (alexkim205) [inline frontend/src/lib/components/DefinitionPopup/DefinitionPopup.tsx:232]: console 🪵

File: frontend/src/lib/components/PropertyKeyInfo.tsx [hunk 4]
```diff
@@ -569,7 +569,15 @@ export function PropertyKeyTitle({ data }: { data: KeyMapping }): JSX.Element {
     )
 }
 
-export function PropertyKeyDescription({ data, value }: { data: KeyMapping; value: string }): JSX.Element {
+export function PropertyKeyDescription({
+    data,
+    value,
+    item,
```
Comment (alexkim205) [inline frontend/src/lib/components/PropertyKeyInfo.tsx:575]: nit: pass in specific "type" instead of entire "item" object.

**Review by alexkim205 — CHANGES_REQUESTED** [review]
This is cool! Left some inline comments and one q below?

- Should we also expose this to non-ee users? It seems like we already expose property type in the tables to non-ee users already.

File: frontend/src/lib/components/DefinitionPopup/DefinitionPopup.tsx (hunk 3 above)
Comment (pauldambra) [reply frontend/src/lib/components/DefinitionPopup/DefinitionPopup.tsx:232]: 🤦

File: frontend/src/lib/components/PropertiesTable.scss (hunk 1 above)
Comment (pauldambra) [reply frontend/src/lib/components/PropertiesTable.scss]: managed to remove it entirely 👍

**Comment by pauldambra** [comment]
Put the type on the right... 

In the new taxonomic view this meant truncating the value sometimes. In the old pop-up letting it roll onto a new line

@alexkim205 is the intention to remove the old pop-up? Which would suggest different behaviour between them isn't too bad?

second-run-property-types

## flag on

Screenshot 2022-02-15 at 10 28 23

## flag off

Screenshot 2022-02-15 at 11 23 42

**Review by clarkus — APPROVED** [review]
I think the right-aligned type will work for most cases. The long, unbreakable strings are problematic regardless, so I'd rather not optimize for that case. These cards are really just quick summaries of the basic qualitative attributes. A user can always click through to see the full details if they're needing more characters.

**Comment by alexkim205** [comment]
> @alexkim205 is the intention to remove the old pop-up? Which would suggest different behaviour between them isn't too bad?

Yup the old popup should eventually be replaced with the new one appwide. Everything else lgtm!

File: frontend/src/lib/components/TaxonomicFilter/InfiniteList.tsx [hunk 5]
```diff
@@ -174,11 +174,14 @@ const renderItemPopupWithoutTaxonomy = (
 
         if (data) {
             return (
-                <div style={{ width, overflowWrap: 'break-word' }}>
+                <div style={{ width, overflowWrap: 'break-word' }} className={'item-popup'}>
```
Comment (alexkim205) [inline frontend/src/lib/components/TaxonomicFilter/InfiniteList.tsx:177]: I think `className` here isn't being used anymore

**Review by alexkim205 — APPROVED** [review]

**Pull request merged**